### PR TITLE
New: Add tags field to MovieCollection

### LINKED ISSUE #8492: Add Tags to all movies if adding a collection
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Is your feature request related to a problem? Please describe

I'm not sure if it is intended or not, but I see it more as a "feature" request or functionality request and not as a bug.

If you add a single movie to the database from the search field and select "movie & collection" as monitoring status, the tags you select for that specific item are not applied to the other movies that are added from that collection

### Describe the solution you'd like

If adding any movie with/as a collection, apply the selected tags to all movies of that collection, rather than just to the one you added from the search field

### Describe alternatives you've considered

Alternative is to add the desired tags 1 by 1 to each movie of the collection manually

### Anything else?

Nope =) Let me know if the current behaviour is intended, then all is fine <3

### PR DESCRIPTION
**Pull request by ricci2511**
#### Database Migration
YES

#### Description
Added a Tags column to the Collections table and corresponding field to MovieCollection.cs, allowing for the handling of tags within a collection, rather than solely on individual movies.

#### Issues Fixed or Closed by this PR
* Closes #8492

### CODE REVIEW

**Comment by ricci2511** [comment]
Based on feedback from the discord chat I will add more changes to this.

Specifically adding some functionality to prepopulate the tags input field when adding a new movie that belongs to a collection that has tags.

**Comment by Qstick** [comment]
This looks good, the one add I would do is allowing setting and changing of tags in `EditCollectionModalContent` in the UI

File: src/NzbDrone.Core/Datastore/Migration/223_add_tags_to_collections.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+using FluentMigrator;
+using NzbDrone.Core.Datastore.Migration.Framework;
+
+namespace NzbDrone.Core.Datastore.Migration
+{
+    [Migration(223)]
```
Comment (Qstick) [inline src/NzbDrone.Core/Datastore/Migration/223_add_tags_to_collections.cs:6]: 224, 223 is already taken by another PR

**Comment by ricci2511** [comment]
Added UI for editing and made changes to the API, particularly in the UpdateCollection method.

The update of the API includes the implementation of updating tags for existing collection movies to reflect newly added collection tags. However, it intentionally excludes the removal of tags from the collection to prevent overriding any tags that users may have added to specific collection movies. More info on this is in the Discord chat.

There just one thing that we would need to handle, and that is the client state updating of movies whose tags have been added to. I would appreciate a suggestion to come up with a solution for that.

There is just one thing we'd still need to handle - updating the client state for movies whose tags have been added. Any suggestions or insights on how to efficiently handle the client-side update of movies reflecting the newly added tags would be greatly appreciated. @Qstick

**Comment by Qstick** [comment]
Pushing master today, then will get this in.

**Review by Qstick — APPROVED** [review]

**Pull request merged**